src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -311,7 +311,7 @@ public static boolean startsWithChar(String value, char prefix) {
      *     {@code false} otherwise.
      */
     public static boolean endsWithChar(String value, char suffix) {
-        return !value.isEmpty() && value.charAt(value.length() - 1) == suffix;
+        return value.charAt(value.length() - 1) == suffix;
     }
```

Comment:
@rdiachenko really sorry for delay 

After the https://github.com/checkstyle/checkstyle/pull/13427#discussion_r1278549204 The removal of this will not create issue because value will not be empty ever

Reply:
Could you regenerate report please?

Reply:
Yes, thanks for remembering It slips from my mind

I will generate a report. and provide all the links

Reply:
@Kevin222004 , what classes use this method? 
Your update makes it very exception prone.
And maybe we should just use function from some existing utilities library.
We don't need personal string until functions, if some library that we already use has such function , we should reuse it

Reply:
Reusage of function from utilities library eliminate requirements to test such code.

Reply:
https://github.com/checkstyle/checkstyle/pull/13427#discussion_r1290185472

> what classes use this method?

There are 2 checks which use this method 
1) MagicNumberCheck :- https://checkstyle.org/checks/coding/magicnumber.html#MagicNumber 
2) ExplicitInitializationCheck :- https://checkstyle.org/checks/coding/explicitinitialization.html 

> we should just use function from some existing utilities library.

I have tried to find but I have not find anything still I will look for it.

Reply:
acocrding to javadoc it was added to fix pmd rule, but currently jvm do not have this performance problems, so even rule was marked for removal  - https://github.com/pmd/pmd/issues/2740
I think we need to remove this method also, if pmd complains, disable this rule, as it is will be removed very soon.

Reply:
I rechecked code and all usage as with hardcoded charater, so all characters should be changed to strings with one symbol in it.

Reply:
@romani Sorry I think I have missed something

https://github.com/checkstyle/checkstyle/pull/13427#discussion_r1290185472

>  what classes use this method?

The method `endsWithChar` is used by following classes 
1) PackageNamesLoader
2) XMLLogger.java
3) UpperEllCheck
4) ImportOrderCheck
5) JavadocMethodCheck
6) CheckUtil

I think the similar one method 
https://github.com/checkstyle/checkstyle/blob/8134680099d82c13c8231ca1b96048a01773a356/src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java#L292-L294
which also similar to this 
https://github.com/checkstyle/checkstyle/blob/8134680099d82c13c8231ca1b96048a01773a356/src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java#L280-L283

should i remove this also

Reply:
https://github.com/checkstyle/checkstyle/pull/13427#discussion_r1290189176

> Reusage of function from utilities library eliminate requirements to test such code.

I am sorry but for which method are you talking about `parseDouble` --> in checkutil 
or 
`endsWithChar` -> in commonutil 

if this is `parseDouble` -> then I still not find anything which can replace this in 

if it is about `endsWithChar` -> in commonutil 
Then i have removed both method from util
https://github.com/checkstyle/checkstyle/blob/8134680099d82c13c8231ca1b96048a01773a356/src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java#L292
and 
https://github.com/checkstyle/checkstyle/blob/8134680099d82c13c8231ca1b96048a01773a356/src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java#L313

please check

Reply:
endsWithChar startsWithChar should be replaced to String.startWith and String.endsWith methods that are in jdk.

Reply:
@romani I have made this changes

But any opinion on failure of run-inspection https://app.circleci.com/pipelines/github/checkstyle/checkstyle/20172/workflows/0a1684f1-be1f-4539-a24a-9fc88acb28ed/jobs/359238/artifacts :)

Reply:
Disable it at https://github.com/checkstyle/checkstyle/blob/8134680099d82c13c8231ca1b96048a01773a356/config/intellij-idea-inspections.xml#L4054
With comment

Reply:
done